<?xml version="1.0"?>

<launch>
	<arg name="robotName" value="intel_aero"/>
	<arg name="controller" value="pid"/> <!-- pid, si_t1_flc_fm, di_t1_flc_fm, si_it2_flc_fm, di_it2_flc_fm -->
	<arg name="noise" default="0.00"/>

	<remap from="/uav/command" to="/$(arg robotName)/command"/>
	<remap from="/uav/command_camera" to="/$(arg robotName)/command_camera"/>
	<remap from="/uav/command_velocity" to="/$(arg robotName)/command_velocity"/>
	<remap from="/uav/command_velocity_keyboard" to="/$(arg robotName)/command_velocity"/>
	<remap from="/uav/position" to="/$(arg robotName)/position"/>
	<remap from="/uav/velocity" to="/$(arg robotName)/linear_velocity"/>
	<remap from="/uav/noise/position" to="/$(arg robotName)/noise/position"/>
	<remap from="/uav/odometry" to="/vins_estimator/odometry"/>
	<remap from="/uav/true_odometry" to="/$(arg robotName)/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/$(arg robotName)/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/$(arg robotName)/trajectory_velocity"/>
	<remap from="/uav/noise" to="/$(arg robotName)/noise"/>
	<remap from="/uav/pose" to="/$(arg robotName)/command/pose"/>

	<include file="$(find controllers)/launch/$(arg controller).launch"/> 

	<node name="trajectory" pkg="controllers" type="trajectory" output="screen" args="none"/>
	<!-- node name="noise" pkg="controllers" type="noise" output="screen" args="$(arg noise)"/ -->
	<!-- node name="logger" pkg="controllers" type="logger" output="screen" args="$(arg controller)"/ -->

	<node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/>
</launch>
